src/java.base/share/classes/java/util/jar/JarVerifier.java

```diff
@@ -145,7 +144,7 @@ public void beginEntry(JarEntry je, ManifestEntryVerifier mev)
                 }
 
                 if (uname.equals(JarFile.MANIFEST_NAME) ||
-                        uname.equals(JarIndex.INDEX_NAME)) {
+                        uname.equals(JarFile.INDEX_NAME)) {
```

Comment:
It would be useful if someone from security-libs could comment on this. The interaction between signed JAR and JAR index isn't very clear. The change you have is safe but it might be that we can drop the checking for INDEX.LIST here.

Reply:
I am thinking this line should not be removed for compatibility with existing JARs that have indexes.

Reply:
still keep the code